apps/src/templates/instructions/InlineAudio.jsx

```diff
@@ -0,0 +1,125 @@
+import Radium from 'radium';
+import React from 'react';
+import experiments from '../../experiments';
+
+import { connect } from 'react-redux';
+
+const styles = {
+  error: {
+    display: 'inline-block',
+    marginLeft: 10,
+    marginBottom: 0,
+    padding: '5px 10px'
+  },
+
+  button: {
+    'float': 'left',
+    border: 'none',
+    height: 32,
+    margin: 0,
+    outline: 'none',
+    padding: 8,
+    width: 33,
+    boxSizing: 'border-box'
+  },
+
+  volumeButton: {
+    background: "#7664A0",
+    borderRadius: "100px 0px 0px 100px",
+  },
+
+  playPauseButton: {
+    background: "#A69BC1",
+    borderRadius: "0px 100px 100px 0px",
+  },
+
+  buttonImg: {
+    opacity: 1,
+    'float': 'left'
+  }
+};
+
+const InlineAudio = React.createClass({
+  propTypes: {
+    assetUrl: React.PropTypes.func.isRequired,
+    src: React.PropTypes.string.isRequired
+  },
+
+  componentWillUpdate: function (nextProps) {
+    if (this.props.src !== nextProps.src) {
+      // unload current Audio object
+      var audio = this.state.audio;
+      audio.src = undefined;
+      audio.load();
+
+      this.setState({
+        audio: undefined,
+        playing: false,
+      });
+    }
+  },
+
+  getInitialState: function () {
+    return {
+      audio: undefined,
+      playing: false,
+    };
+  },
+
+  getAudioElement: function () {
+    if (this.state.audio) {
+      return this.state.audio;
+    }
+
+    var audio = new Audio(this.props.src);
+    $(audio).on("ended", e => {
```

Comment:
ah, yes, and also an unnecessary one; I can replace this with a simple addEventListener